### ISSUE TITLE
Install Third Party script fails on double-conversion.build

### ISSUE THREAD

**Issue by jasonsturges** [facebook/react-native#21354]
## Environment

Environment:
  OS: macOS High Sierra 10.13.6
  Node: 10.11.0
  Yarn: 1.10.1
  npm: 6.4.1
  Watchman: 4.9.0
  Xcode: Xcode 10.0 Build version 10A255
  Android Studio: Not Found

Packages: (wanted => installed)
  react: 16.0.0 => 16.0.0
  react-native: 0.51.0 => 0.51.0

## Description

On build step, running script 'Install Third Party' fails when executing 'double-conversion.build' build command:

> PhaseScriptExecution Install\ Third\ Party /Users/jsturges/Library/Developer/Xcode/DerivedData/realsimple-apzvagrtoaeanyfegvqrfvpgmyko/Build/Intermediates.noindex/ArchiveIntermediates/realsimple/IntermediateBuildFilesPath/React.build/Release-iphoneos/double-conversion.build/Script-190EE32F1E6A43DE00A8543A.sh

In my `~/.rncache`, I have:
- boost_1_63_0.tar.gz
- double-conversion-1.1.5.tar.gz
- folly-2016.09.26.00.tar.gz
- glog-0.3.4.tar.gz

There's a lot of chatter here regarding this, but many issues cite having the library within my `.rncache`.  

Per Issue #14447, I've tried creating a `third-party` folder in: `node_modules/react-native/third-party` and placing libraries from my `~/.rncache` there unzipped.  Then, I need to manually configure glog.

```
$ cd node_modules/react-native/third-party/glog-0.3.4
$ ./configure
```

That results in error:
>▸ Compiling signalhandler.cc
>❌ node_modules/react-native/third-party/glog-0.3.4/src/signalhandler.cc:78:28: no member named '__rip' in '__darwin_arm_thread_state'
> `return (void*)context->PC_FROM_UCONTEXT;`

Per Issue #16106, I can alter `signalhandler.cc`:

```
$ vi node_modules/react-native/third-party/glog-0.3.4/src/signalhandler.cc
```

At Line 78, change: `return (void*)context->PC_FROM_UCONTEXT;` to `return NULL;`

Which then fails on: Building library libRCTWebSocket.a

> ❌  error: Build input file cannot be found: 'node_modules/react-native/Libraries/WebSocket/libfishhook.a'

Per Issue #16039, I can't get past changing the "fishhook.h" import.

Are there known issues with my environment, as this project runs on numerous other MacBooks.

## Reproducible Demo

This is within the build, executing: `npm run build`

**Comment by hramos** [comment]
Duplicate of #20774

**Comment by hramos** [comment]
You're using Xcode 10, which has some known issues (#19573), many of which only have been fixed in 0.56 and 0.57, but you're still using 0.51.

Upgrading to 0.57 will help a lot here, but you may still be blocked by #20774. Falling back to the old build system might help.

**Comment by cmelendez** [comment]
Open `MyProject.xcworkspace` instead of `MyProject. xcodeproj`.

After struggling for hours and following every possible recommendation this did it for me.
